modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerConfigurationChangesTest.java

```diff
@@ -120,21 +125,28 @@ public void setUp() throws Exception {
 
         metaStorageManager.put(zonesLogicalTopologyVersionKey(), longToBytes(0));
 
+        ConfigurationRegistry clusterConfigRegistry = clusterCfgMgr.configurationRegistry();
+
+        Consumer<Function<Long, CompletableFuture<?>>> registry =
+                c -> clusterConfigRegistry.listenUpdateStorageRevision(c::apply);
+
         distributionZoneManager = new DistributionZoneManager(
                 zonesConfiguration,
                 tablesConfiguration,
                 metaStorageManager,
                 logicalTopologyService,
                 vaultMgr,
+                registry,
                 "test"
         );
 
         vaultMgr.start();
         clusterCfgMgr.start();
         metaStorageManager.start();
-        distributionZoneManager.start();
 
-        metaStorageManager.deployWatches();
+        deployWatchesAndUpdateMetaStorageRevision(metaStorageManager);
```

Comment:
Why do we deploy watches before starting the distributionZoneManager? It seems incorrect, because common practice is to deploy watches after all components start up.